doc/mgmt/OpenConfig_Path_Translation_Root_Expansion_HLD.md

```diff
@@ -0,0 +1,338 @@
+
+###### Single Root Delete
+It will be translated into multiple delete paths as show below:
+```
+Original Delete
+prefix:<origin:"openconfig">delete:<>
+```
+```
+Translated Delete
+prefix:<>delete:<elem:<name:"openconfig-interfaces:interfaces">>delete:<elem:<name:"openconfig-lacp:lacp">>delete:<elem:<name:"openconfig-platform:components">>delete:<elem:<name:"openconfig-qos:qos">>delete:<elem:<name:"openconfig-sampling:sampling">>delete:<elem:<name:"openconfig-system:system">>
+```
+
+###### Single Root Update/Replace
+For update and replace, the JSON payload needs to be parsed and the payload has to be separated into multiple JSON payloads according to the top-container names, and then placed into individual updates/replaces.
+```
+Original Update
+prefix:<origin:"openconfig">
+update:<path:<> val:<json_ietf_val:
+"{
+  \"openconfig-interfaces:interfaces:\": {...},
+  \"openconfig-platform:components:\": {xxx}
+}“
+>>
+```
+```
+Translated Update
+prefix:<>
+update:<path:<elem:<name:"openconfig-interface:interfaces">> val:<json_ietf_val:
+"{
+  \"openconfig-interfaces:interfaces:\": {...},
+}“>
+update:<path:<elem:<name:"openconfig-platform:components">> val:<json_ietf_val:
+"{
+  \"openconfig-platform:components:\": {...},
+}“>
+>
+```
+
+##### SetResponse
+Since we do not support the mix of single root set and non-single root set operations, we will only translate single root SetResponse if the original SetRequest is single root.
+For the translation, we just need to use a single root UpdateResult to replace the Delete/Update/Replace.
+For example:
+```
+Original SetResponse
+prefix:<>
+response:<timestamp:123456 path:<elem:<name:"openconfig-interfaces:interfaces">>op:Update>
+response:<timestamp:123456 path:<elem:<name:"openconfig-platform:components">>op:Update>
+response:<timestamp:123456 path:<elem:<name:"openconfig-qos:qos">>op:Replace>
+response:<timestamp:123456 path:<elem:<name:"openconfig-lacp:lacp">>op:Replace>
+```
+```
+Translated SetResponse
+prefix:<origin:"openconfig">
+response:<timestamp:123456 path:<>op:Update>
+response:<timestamp:123456 path:<>op:Replace>
+```
+
+##### SubscribeRequest
+For a single root SubscribeRequest, the translation is similar to a GetRequest. It will be translated into a SubscribeRequest with multiple top-container paths.
+
+##### SubscribeResponse
+Since SubscribeRespone is per update, there is nothing to change for single root translation on updates.
+
+
+### SAI API
+
+This HLD does not include any changes to the SAI API.
+
+### Configuration and Management
+The configuration is updated with the full set of OpenConfig models at the root, instead of individual subtrees.
+
+#### CLI/YANG model Enhancements
+
+This HLD does not include any enhancements to CLI or YANG models.
+
+#### Config DB Enhancements
+
+This HLD does not include any changes to the Config DB.
+
+### Warmboot and Fastboot Design Impact
+
+This HLD does not have any impact on Warmboot and Fastboot.
+
+### Memory Consumption
```

Comment:
by supporting the root level path for both get / subscribe may cause out of memory issue in large scale setup, we may need more info?